src/index.tsx

```diff
@@ -35,12 +35,22 @@ function concatTranscripts(...parts: string[]) {
     .trim();
 }
 
-export function useSpeechRecognition(options: SpeechRecognitionOptions = defaultOptions): SpeechRecognitionUtils {
+export function useSpeechRecognition(
+  userOptions: Partial<SpeechRecognitionOptions> = defaultOptions,
+): SpeechRecognitionUtils {
   const [
     { error, status, pauseAfterRecognitionEnd, interimTranscript, finalTranscript, transcript },
     dispatch,
   ] = useReducer<ReducerBuilder<SpeechRecognitionInternalState>>(speechRecognitionReducer, initialState);
 
+  const options = useMemo(
+    () => ({
+      ...defaultOptions,
+      ...userOptions,
+    }),
+    [userOptions],
+  );
+
```

Comment:
This is because I wanted all options to be optional ,so I decided to just merge the default options and the userOptions, that way we don't need to set all properties if we don't need them

Reply:
I had planned to do this at some point. Thanks for handling it!